



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.3.0">
    
    
      
        <title>Whole Metagenome Sequencing - Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.572ca0f0.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.22915126.css">
      
      
        
        
        <meta name="theme-color" content="#e91e63">
      
    
    
      <script src="../assets/javascripts/modernizr.962652e9.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="pink" data-md-color-accent="pink">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#whole-metagenome-sequencin" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Bioinformatics Tutorials
              </span>
              <span class="md-header-nav__topic">
                Whole Metagenome Sequencing
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href=".." title="Bioinformatics Tutorials" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bioinformatics Tutorials
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Individual lessons
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Individual lessons
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control and Trimming" class="md-nav__link">
      Quality Control and Trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mapping/" title="Mapping and Variant Calling" class="md-nav__link">
      Mapping and Variant Calling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-nav__link">
      Pan-Genome Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../16S/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Whole Metagenome Sequencing
      </label>
    
    <a href="./" title="Whole Metagenome Sequencing" class="md-nav__link md-nav__link--active">
      Whole Metagenome Sequencing
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" title="Table of Contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#microbiome-used" title="Microbiome used" class="md-nav__link">
    Microbiome used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#whole-metagenome-sequencing" title="Whole Metagenome Sequencing" class="md-nav__link">
    Whole Metagenome Sequencing
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#softwares-required-for-this-tutorial" title="Softwares Required for this Tutorial" class="md-nav__link">
    Softwares Required for this Tutorial
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-and-organise-your-working-directory" title="Prepare and organise your working directory" class="md-nav__link">
    Prepare and organise your working directory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-the-data-and-checking-their-quality" title="Getting the Data and Checking their Quality" class="md-nav__link">
    Getting the Data and Checking their Quality
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-classification" title="Taxonomic Classification" class="md-nav__link">
    Taxonomic Classification
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization-with-pavian" title="Visualization with Pavian" class="md-nav__link">
    Visualization with Pavian
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../meta_assembly_2/" title="Metagenome assembly (continued)" class="md-nav__link">
      Metagenome assembly (continued)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../metavir/" title="Viral metagenomics" class="md-nav__link">
      Viral metagenomics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../nanopore/" title="Introduction to Nanopore Sequencing" class="md-nav__link">
      Introduction to Nanopore Sequencing
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" title="Table of Contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#microbiome-used" title="Microbiome used" class="md-nav__link">
    Microbiome used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#whole-metagenome-sequencing" title="Whole Metagenome Sequencing" class="md-nav__link">
    Whole Metagenome Sequencing
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#softwares-required-for-this-tutorial" title="Softwares Required for this Tutorial" class="md-nav__link">
    Softwares Required for this Tutorial
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-and-organise-your-working-directory" title="Prepare and organise your working directory" class="md-nav__link">
    Prepare and organise your working directory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-the-data-and-checking-their-quality" title="Getting the Data and Checking their Quality" class="md-nav__link">
    Getting the Data and Checking their Quality
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-classification" title="Taxonomic Classification" class="md-nav__link">
    Taxonomic Classification
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization-with-pavian" title="Visualization with Pavian" class="md-nav__link">
    Visualization with Pavian
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/HadrienG/tutorials/edit/master/docs/wms.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="whole-metagenome-sequencin">Whole Metagenome Sequencin<a class="headerlink" href="#whole-metagenome-sequencin" title="Permanent link">&para;</a></h1>
<h3 id="table-of-contents">Table of Contents<a class="headerlink" href="#table-of-contents" title="Permanent link">&para;</a></h3>
<ul>
<li><a href="#introduction">Introduction</a><ul>
<li><a href="#the-pig-microbiome">The Pig Microbiome</a></li>
<li><a href="#whole-metagenome-sequencing">Whole Metagenome Sequencing</a></li>
</ul>
</li>
<li><a href="#softwares-required-for-this-tutorial">Softwares Required for this Tutorial</a></li>
<li><a href="#getting-the-data-and-checking-their-quality">Getting the Data and Checking their Quality</a></li>
<li><a href="#taxonomic-classification">Taxonomic Classification</a></li>
<li><a href="#visualization">Visualization</a></li>
</ul>
<h3 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h3>
<h4 id="microbiome-used">Microbiome used<a class="headerlink" href="#microbiome-used" title="Permanent link">&para;</a></h4>
<p>In this tutorial we will compare samples from the Pig Gut Microbiome to samples from the Human Gut Microbiome. Below you'll find a brief description of the two projects:</p>
<p>The Pig Microbiome:</p>
<blockquote>
<p>Pig is a main species for livestock and biomedicine. The pig genome sequence was recently reported. To boost research, we established a catalogue of the genes of the gut microbiome based on faecal samples of 287 pigs from France, Denmark and China. More than 7.6 million non-redundant genes representing 719 metagenomic species were identified by deep metagenome sequencing, highlighting more similarities with the human than with the mouse catalogue. The pig and human catalogues share only 12.6 and 9.3 % of their genes, respectively, but 70 and 95% of their functional pathways. The pig gut microbiota is influenced by gender, age and breed. Analysis of the prevalence of antibiotics resistance genes (ARGs) reflected antibiotics supplementation in each farm system, and revealed that non-antibiotics-fed animals still harbour ARGs. The pig catalogue creates a resource for whole metagenomics-based studies, highly valuable for research in biomedicine and for sustainable knowledge-based pig farming</p>
</blockquote>
<p>The Human Microbiome:</p>
<blockquote>
<p>We are facing a global metabolic health crisis provoked by an obesity epidemic. Here we report the human gut microbial composition in a population sample of 123 non-obese and 169 obese Danish individuals. We find two groups of individuals that differ by the number of gut microbial genes and thus gut bacterial richness. They harbour known and previously unknown bacterial species at different proportions; individuals with a low bacterial richness (23% of the population) are characterized by more marked overall adiposity, insulin resistance and dyslipidaemia and a more pronounced inflammatory phenotype when compared with high bacterial richness individuals. The obese individuals among the former also gain more weight over time. Only a few bacterial species are sufficient to distinguish between individuals with high and low bacterial richness, and even between lean and obese. Our classifications based on variation in the gut microbiome identify subsets of individuals in the general white adult population who may be at increased risk of progressing to adiposity-associated co-morbidities</p>
</blockquote>
<h4 id="whole-metagenome-sequencing">Whole Metagenome Sequencing<a class="headerlink" href="#whole-metagenome-sequencing" title="Permanent link">&para;</a></h4>
<p>Whole Metagenome sequencing (WMS), or shotgun metagenome sequencing, is a relatively new and powerful sequencing approach that provides insight into community biodiversity and function. On the contrary of Metabarcoding, where only a specific region of the bacterial community (the 16s rRNA) is sequenced, WMS aims at sequencing all the genomic material present in the environment.</p>
<p>The choice of shotgun or 16S approaches is usually dictated by the nature of the studies being conducted. For instance, 16S is well suited for analysis of large number of samples, i.e., multiple patients, longitudinal studies, etc. but offers limited taxonomical and functional resolution. WMS is generally more expensive but offers increased resolution, and allows the discovery of viruses as well as other mobile genetic elements.</p>
<h3 id="softwares-required-for-this-tutorial">Softwares Required for this Tutorial<a class="headerlink" href="#softwares-required-for-this-tutorial" title="Permanent link">&para;</a></h3>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a href="https://ccb.jhu.edu/software/kraken/">Kraken</a></li>
<li><a href="https://www.r-project.org/">R</a></li>
<li><a href="https://github.com/fbreitwieser/pavian">Pavian</a></li>
</ul>
<h3 id="prepare-and-organise-your-working-directory">Prepare and organise your working directory<a class="headerlink" href="#prepare-and-organise-your-working-directory" title="Permanent link">&para;</a></h3>
<p>You will first login to your virtual machine using the IP provided by the teachers.
All the exercise will be performed on your VM in the cloud.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When you login with the ssh command, please add the option -X at the end of it to be able to use graphical interface</p>
</div>
<div class="codehilite"><pre><span></span>mkdir ~/wms
<span class="nb">cd</span> ~/wms
mkdir data
mkdir results
mkdir scripts
</pre></div>


<h3 id="getting-the-data-and-checking-their-quality">Getting the Data and Checking their Quality<a class="headerlink" href="#getting-the-data-and-checking-their-quality" title="Permanent link">&para;</a></h3>
<p>As the data were very big, we have prepared performed a downsampling on all 6 datasets (3 pigs and 3 humans).
We will first download and unpack the data.</p>
<div class="codehilite"><pre><span></span><span class="nb">cd</span> ~/wms/data
curl -O -J -L https://osf.io/h9x6e/download
tar xvf subset_wms.tar.gz
<span class="nb">cd</span> sub_100000
</pre></div>


<p>We'll use FastQC to check the quality of our data.
FastQC should be already installed on your VM, so you need to type</p>
<div class="codehilite"><pre><span></span>fastqc *.fastq
</pre></div>


<p>If the quality appears to be good, it's because it was probably the cleaned reads that were deposited into SRA.
We can directly move to the classification step.</p>
<h3 id="taxonomic-classification">Taxonomic Classification<a class="headerlink" href="#taxonomic-classification" title="Permanent link">&para;</a></h3>
<p><a href="https://ccb.jhu.edu/software/kraken/">Kraken</a> is a system for assigning taxonomic labels to short DNA sequences (i.e. reads)<br />
Kraken aims to achieve high sensitivity and high speed by utilizing exact alignments of k-mers and a novel classification algorithm (sic).</p>
<p>In short, kraken uses a new approach with exact k-mer matching to assign taxonomy to short reads. It is <em>extremely</em> fast compared to traditional
approaches (i.e. BLAST).</p>
<p>By default, the authors of kraken built their database based on RefSeq Bacteria, Archaea and Viruses. We'll use it for the purpose of this tutorial.
We will download a shrunk database (minikraken) provided by Kraken developers that is only 4GB.</p>
<div class="codehilite"><pre><span></span><span class="c1"># First we create a databases directory in our home</span>
<span class="nb">cd</span> /mnt
sudo mkdir databases
<span class="nb">cd</span> databases
<span class="c1"># Then we download the minikraken database</span>
sudo wget https://ccb.jhu.edu/software/kraken/dl/minikraken_20171019_4GB.tgz
sudo tar xzf minikraken_20171019_4GB.tgz
<span class="nv">KRAKEN_DB</span><span class="o">=</span>/mnt/databases/minikraken_20171013_4GB
<span class="nb">cd</span>
</pre></div>


<p>Now run kraken on the reads</p>
<div class="codehilite"><pre><span></span><span class="c1"># In the data/ directory</span>
<span class="nb">cd</span> ~/wms/data/sub_100000
<span class="k">for</span> i in *_1.fastq
<span class="k">do</span>
    <span class="nv">prefix</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$i</span> _1.fastq<span class="k">)</span>
    <span class="c1"># print which sample is being processed</span>
    <span class="nb">echo</span> <span class="nv">$prefix</span>
    kraken --db <span class="nv">$KRAKEN_DB</span> --threads <span class="m">2</span> --fastq-input <span class="se">\</span>
        <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_1.fastq <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_2.fastq &gt; /home/student/wms/results/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.tab
    kraken-report --db <span class="nv">$KRAKEN_DB</span> <span class="se">\</span>
        /home/student/wms/results/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.tab &gt; /home/student/wms/results/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_tax.txt
<span class="k">done</span>
</pre></div>


<p>which produces a tab-delimited file with an assigned TaxID for each read.</p>
<p>Kraken includes a script called <code>kraken-report</code> to transform this file into a "tree" view with the percentage of reads assigned to each taxa. We've run this script at each step in the loop. Take a look at the <code>_tax.txt</code> files!</p>
<h3 id="visualization-with-pavian">Visualization with Pavian<a class="headerlink" href="#visualization-with-pavian" title="Permanent link">&para;</a></h3>
<p>Pavian is a web application for exploring metagenomics classification results.</p>
<p>Install and run Pavian from R:</p>
<div class="codehilite"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repos</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="n">CRAN</span> <span class="o">=</span> <span class="s">&quot;http://cran.rstudio.com&quot;</span><span class="p">))</span>
<span class="nf">if </span><span class="p">(</span><span class="o">!</span><span class="nf">require</span><span class="p">(</span><span class="n">remotes</span><span class="p">))</span> <span class="p">{</span> <span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;remotes&quot;</span><span class="p">)</span> <span class="p">}</span>
<span class="n">remotes</span><span class="o">::</span><span class="nf">install_github</span><span class="p">(</span><span class="s">&quot;fbreitwieser/pavian&quot;</span><span class="p">)</span>
<span class="n">pavian</span><span class="o">::</span><span class="nf">runApp</span><span class="p">(</span><span class="n">port</span><span class="o">=</span><span class="m">5000</span><span class="p">)</span>
</pre></div>


<p>Then you will explore and compare the results produced by Kraken.</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../16S/" title="Metabarcoding" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Metabarcoding
              </span>
            </div>
          </a>
        
        
          <a href="../meta_assembly/" title="Metagenome assembly" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Metagenome assembly
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../assets/fonts/font-awesome.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.a353778b.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:".."}})</script>
      
    
  </body>
</html>